### PR TITLE
Modify main.py and add a json file

### PR DESCRIPTION
**Pull request by tokinoterumi**
## What

Modified the listing api app and created a json file containing listing items data

## CHECKS :warning:

Please make sure you are aware of the following.

- [x] **The changes in this PR doesn't have private information

### CODE REVIEW

File: python/main.py [hunk 1]
```diff
@@ -18,18 +21,54 @@
...
 
+@app.get("/items")
+def listed_items():
+    return items
+
+@app.get("/items/{id}")
+def item_details(id: str):
+    for item in items_list:
+        if item["id"] == id:
+            return item
```
Comment (roryhow) [inline python/main.py:74]: What about if there are no matches in this list? currently the server responds a `null` response with `200 OK` status code. 

Is a status code that might be more suitable for this case?

File: python/main.py [hunk 2]
```diff
@@ -18,18 +21,54 @@
...
+
+    new_item = {
+            "id": id,
+            "name" : name,
+            "category" : category,
+            "image" : new_image_name
+            }
+
+    items_list.append(new_item)
+    #items.append(new_item)
```
Comment (roryhow) [inline python/main.py]: Not super important, but you could remove this line if you don't need it

File: python/main.py [hunk 3]
```diff
@@ -18,18 +21,54 @@
...
 def root():
     return {"message": "Hello, world!"}
 
 @app.post("/items")
-def add_item(name: str = Form(...)):
+async def add_item(name: str = Form(...),
+                category: str = Form(...),
+                image: UploadFile = File(...)):
+    image_bytes = await image.read() # turn the image into a buffer-like object
+    image_hashed = hashlib.sha256(image_bytes).hexdigest()  # hash the image
```
Comment (roryhow) [inline python/main.py:45]: How do we know if this is a valid image at this point? What happens if you try to upload a `.txt` or `.json` file?

Comment (tokinoterumi) [reply python/main.py:45]: fixed! I added following codes.

    ALLOWED_IMAGE_EXTENSIONS = ['.jpg', '.jpeg', '.png']
    def is_image_file(filename: str) -> bool:
        extension = os.path.splitext(filename)[1].lower()
        return extension in ALLOWED_IMAGE_EXTENSIONS
    if not is_image_file(image.filename):
        return {"error": "Invalid image file"}

File: python/main.py (hunk 1 above)
Comment (tokinoterumi) [reply python/main.py:74]: fixed!

File: python/main.py (hunk 3 above)
Comment (roryhow) [reply python/main.py:45]: Nice! Similar to the another comment I made, this message `{"error": "Invalid image file"}` will be given a 200 OK status code.

Is there a status code that might be more suitable for this situation?

Comment (tokinoterumi) [reply python/main.py:45]: fixed!

**Review by roryhow — APPROVED** [review]

**Pull request merged**